### DISCUSSION TITLE
Can SX1262 achieve half duplex communication

### DISCUSSION THREAD

**Discussion by camerakang** [jgromes/RadioLib#1231] [General]
Can this chip realize the function of receiving ACK from the receiver after the transmitter has finished sending, so that it can send and receive?

**Comment by K4KDR** [comment]
Yes!   In fact, it works so well that I use an SX1262-based LoRa module for the **Pi-Pico**  (https://www.amazon.com/TOP1-Raspberry-902-930MHz-Modulations-Pico-LoRa-SX1262-915M/dp/B0B1M5C7D2?th=1) as a little "LoRa repeater" which I hope to send up on balloon flights soon.     

Not part of your question, but triple-stacked w/ a GPS module of the same form-factor (https://www.amazon.com/Waveshare-L76B-Module-Raspberry-Support/dp/B096XDXPLH), you end up with a very nice package that can both beacon its position and repeat LoRa packets that it hears!

![arduino-code-dual-core](https://github.com/user-attachments/assets/547dfb56-77af-4b3b-9307-7f49de0330f5)

**Comment by jgromes** [comment]
@camerakang You are asking about two different question. You have titled this "Can SX1262 achieve half duplex communication" - yes, in fact it is the only mode possible. SX126x cannot work in a full duplex mode. It can only either transmit or receive, not both at the same time. That's what's called half duplex.

But in your post you are talking about something else - sending acknowledgments to achieve some (unspecified) level of realiability in the link. That's in the scope of some higher-layer protocol, which RadioLib does not implement.
